Added saving of features

deactivated-acc opened the pull request:
New button in imagedisplay activity prompts for filename and saves
keypoints and descriptors as json

As described in issue #17

deactivated-acc committed:
Added saving of features

New button in imagedisplay activity prompts for filename and saves
keypoints and descriptors as json

jrauhamaa commented:
Since we are only saving data files here, do you think it makes sense to ask the user for file names?

jrauhamaa commented on FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java:
```diff
@@ -17,6 +48,201 @@ protected void onCreate(Bundle savedInstanceState) {
+        try{
+            FileWriter fw = null;
+
+            File keypointsFile = new File(fileDir, fileName+"_keypoints");
```
Is this code duplicated on purpose?

jrauhamaa reviewed (changes requested):
- remove duplicated code
- would be nice if the code would generate automatic filename (e.g. an uuid) instead of asking the user given that the user won't be able to access the files. 
- would be nice if the user got notified of successful save

deactivated-acc committed:
Added loading files necessary for matching

-Added match button on image display activity
-Added dialog with spinner for filenames
-When saving adds filename to storage (for spinner)
-Made stub for matching features that loads files and converts them to
Mats

deactivated-acc committed:
Remove duplicate code + save notification

deactivated-acc commented:
-Added loading files for the matching part
-Removed the duplicate code in saveFeatures
-Added toast messages for save success and failure

@jrauhamaa automatic filename generation isn't really useful right now, but can be easily added when it is necessary. With the known filename, you can easily see which one to match with, which you can't with the automatically generated one.

jorenham commented:
You got some conflict

deactivated-acc committed:
Merge branch 'master' into save-features
